web_client/views/annotationListWidget.js

```diff
@@ -0,0 +1,127 @@
+import _ from 'underscore';
+
+import { AccessType } from 'girder/constants';
+import { confirm } from 'girder/dialog';
+import { getApiRoot } from 'girder/rest';
+import AccessWidget from 'girder/views/widgets/AccessWidget';
+import UserCollection from 'girder/collections/UserCollection';
+import View from 'girder/views/View';
+
+import AnnotationCollection from '../collections/AnnotationCollection';
+
+import annotationList from '../templates/annotationListWidget.pug';
+
+import '../stylesheets/annotationListWidget.styl';
+
+const AnnotationListWidget = View.extend({
+    events: {
+        'change .g-annotation-toggle': '_displayAnnotation',
+        'click .g-annotation-delete': '_deleteAnnotation',
+        'click .g-annotation-permissions': '_changePermissions',
+        'click .g-annotation-row'(evt) {
+            var $el = $(evt.currentTarget);
+            $el.find('.g-annotation-toggle > input').click();
+        },
+        'click .g-annotation-row a,input'(evt) {
+            evt.stopPropagation();
+        }
+    },
+
+    initialize() {
+        this._drawn = new Set();
+        this._viewer = null;
+        this._sort = {
+            'field': 'name',
+            'direction': 1
+        };
+
+        this.collection = this.collection || new AnnotationCollection();
+        this.users = new UserCollection();
+
+        this.listenTo(this.collection, 'all', this.render);
+        this.listenTo(this.users, 'all', this.render);
+
+        this.collection.fetch({
+            itemId: this.model.id,
+            sort: 'created',
+            sortdir: -1
```

Comment:
There is something odd about the display order.  We request the annotations in reverse creation order, but they are displayed in a different order (I'm not sure what).  For instance, I get:
![image](https://user-images.githubusercontent.com/8781639/39262227-d5dbfdfa-488c-11e8-8cea-abba644fd90d.png)
I was expecting those dates to be descending.

Reply:
Is that the order the `/annotations` endpoint returns them in or is the client code changing it for some reason?

Reply:
You might need to change the sorting attributes on the `AnnotationCollection`:
https://github.com/girder/girder/blob/bbab319395d3908f43ac9e0eefaa4fe880338f82/clients/web/src/collections/Collection.js#L17-L18

Reply:
The endpoint returns them in the created date order.  The client is changing that (but not to name order, so I don't know how it is being changed).

Reply:
Well, the default girder behavior is to sort by `model.get('name')`, which is undefined for these models.  If the comparator is stable, it should retain the original order.  I'll just try disabling backbone's sorting and see if that works.